driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java

```diff
@@ -95,7 +93,7 @@ public static Builder builder(final MongoClientSettings settings) {
         private WriteConcern writeConcern = WriteConcern.ACKNOWLEDGED;
         private ReadConcern readConcern = ReadConcern.DEFAULT;
         private CodecRegistry codecRegistry = MongoClients.getDefaultCodecRegistry();
-        private StreamFactoryFactory streamFactoryFactory = createDefaultStreamFactoryFactory();
+        private StreamFactoryFactory streamFactoryFactory;
```

Comment:
A slight change in behavior of this class to not have a default stream factory factory, but it's necessary to not unnecessarily leak an NioEventLoopGroup.